android/src/app/organicmaps/routing/NavigationController.java

```diff
@@ -206,7 +209,7 @@ private void updateVehicle(RoutingInfo info)
     }
   }
 
-  private void updatePedestrian(RoutingInfo info)
+  private void updatePedestrian(@NonNull RoutingInfo info)
   {
     mNextTurnDistance.setText(Utils.formatDistance(mFrame.getContext(), info.distToTurn));
```

Comment:
- Should we do the same stuff here with isValid like in updateVehicle?
- Strange, why do we have separate updateVehicle, updatePedestrian, but no bicycle :)

Reply:
1. No, I removed `isValid` check. We don't need it, as you noted before
2. `updatePedestrian ` runs only for pedestrian, `updateVehicle` runs for all routers.